Makefile

```diff
@@ -9,6 +9,9 @@ CFLAGS := -O2 -g -Wall -Werror $(CFLAGS)
 REPODIR := $(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
 UIDGID := $(shell stat -c '%u:%g' ${REPODIR})
 
+CONTAINER_ENGINE ?= docker
+CONTAINER_RUN_ARGS ?= --user "${UIDGID}"
```

Comment:
Why does podman not need this?

Reply:
podman is daemonless and runs within the context of the current user by default; it doesn't need to be instructed to do so explicitly.

Reply:
https://docs.podman.io/en/latest/markdown/podman-run.1.html#user-u-user-user-group-uid-uid-gid-user-gid-uid-group

If I understood correctly, when Podman is invoked, a rootless container is created and by default a user namespace is used, the UID:GID will then match the user creating the process. However, if they are specified, they could not match.

```
(venv) ~/g/m/ebpf *matt/makefile-podman> CONTAINER_ENGINE=podman CONTAINER_RUN_ARGS= make container-shell
podman run --rm -ti  \
	-v "/home/matt/github/mmat11/ebpf":/ebpf -w /ebpf \
	"quay.io/cilium/ebpf-builder:1631608809"
root@31ea8747c478:/ebpf# touch noargs
root@31ea8747c478:/ebpf# exit
exit
(venv) ~/g/m/ebpf *matt/makefile-podman> ll noargs
-rw-r--r--. 1 matt matt 0 Nov  9 13:41 noargs
```

```
(venv) ~/g/m/ebpf *matt/makefile-podman> sudo setenforce 0
[sudo] password for matt:
(venv) ~/g/m/ebpf *matt/makefile-podman> CONTAINER_ENGINE=podman make container-shell
podman run --rm -ti --user "1000:1000" \
	-v "/home/matt/github/mmat11/ebpf":/ebpf -w /ebpf \
	"quay.io/cilium/ebpf-builder:1631608809"
1000@55b24d3e75c9:~$ touch withargs
touch: cannot touch 'withargs': Permission denied
```